### PR TITLE
Move dashboard checks of unit_in_category to use course offering category

### PR DESCRIPTION
**Pull request by dmcavoy**
### Background

Each year to launch curriculum we have had to coordinate launching between curriculum and engineering because ScriptConstants had to be update in order for features such as the Assignment Dropdown to work. We just recently moved the AssignmentDropdown to no longer rely on ScriptConstants. 

### Change

This PR is the last step to make it so launching curriculum does not require a change to ScriptConstants at the exact same time as published the curriculum. 

It takes `ScriptConstants.unit_in_category?` calls in dashboard and moves them to use `Script.unit_in_category?` which checks the CourseOffering `category` field instead of the ScriptConstants. 

There are still some calls to `ScriptConstants.unit_in_category?` for translations but those do no have to be updated exactly in time with curriculum publishing.

## Links

[Jira](https://codedotorg.atlassian.net/browse/PLAT-1765)

## Testing story

- Existing tests

### CODE REVIEW

**Comment by dmcavoy** [comment]
Need to update fixtures

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -812,19 +808,19 @@ def legacy_curriculum?
...
   end
 
   def hoc?
-    ScriptConstants.unit_in_category?(:hoc, name)
+    Script.unit_in_category?('hoc', name)
   end
 
   def flappy?
-    ScriptConstants.unit_in_category?(:flappy, name)
+    name === 'flappy'
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb]: love this simplification!

also just noticing after i stopped to comment on it that we don't usually use triple equals in ruby, here and above

File: lib/cdo/script_constants.rb [hunk 2]
```diff
@@ -342,11 +342,6 @@ def self.unit_in_category?(category, script)
     return CATEGORIES[category].include? script
   end
```
Comment (davidsbailey) [inline lib/cdo/script_constants.rb:343]: this looks like the right stopping point to me -- the places still using this method are either in pegasus (can't access rails models) or i18n-related. I'll follow up on those when I get to that part of the script constants work.

File: bin/i18n/sync-in.rb [hunk 3]
```diff
@@ -291,7 +291,7 @@ def localize_level_content(variable_strings, parameter_strings)
       # We want to make sure to categorize HoC scripts as HoC scripts even if
       # they have a version year, so this ordering is important
       script_i18n_directory =
-        if ScriptConstants.unit_in_category?(:hoc, script.name)
+        if Script.unit_in_category?('hoc', script.name)
```
Comment (davidsbailey) [inline bin/i18n/sync-in.rb:294]: makes sense that this is safe to use here, because `Script` is accessed earlier 👍

**Review by davidsbailey — APPROVED** [review]
Looks great Dani! Thanks so much for taking the initiative here, in this PR and in this space in general. it's great to see this mess getting untangled!

**Pull request merged**